compiler/rustc_middle/src/mir/terminator.rs

```diff
@@ -250,8 +250,10 @@ pub enum TerminatorKind<'tcx> {
         /// This allows the memory occupied by "by-value" arguments to be
         /// reused across function calls without duplicating the contents.
         args: Vec<Operand<'tcx>>,
-        /// Destination for the return value. If none, the call necessarily diverges.
-        destination: Option<(Place<'tcx>, BasicBlock)>,
+        /// Where the returned value will be written
+        dest_place: Place<'tcx>,
+        /// Where to go after this call returns. If none, the call necessarily diverges.
+        dest_block: Option<BasicBlock>,
```

Comment:
Hmm, I'm not sure. Intuitively, it feels like this information being accessible without having to go looking further might be valuable and simplify things. Maybe not though. I also don't know which version would be better for perf.

In any case, if we do consider such a change, I think we should make a uniform decision for this, `TerminatorKind::SwitchInt` (which currently requires an otherwise block), and `TerminatorKind::InlineAsm` (which does not require a destination/target block) on the basis of the needs of all three. That seems like too big/complicated of a conversation to have for this PR.

Reply:
agreed, let's try this in a follow up